tests/MoBi.Tests/Presentation/Tasks/InitialConditionsTaskSpecs.cs

```diff
@@ -700,13 +713,50 @@ public void the_formula_that_was_refreshed_is_in_the_cache()
          _initialConditionsBuildingBlock.FormulaCache.Contains(_refreshedFormula).ShouldBeTrue();
       }
 
+      [Observation]
+      public void the_added_expression_should_be_renamed()
+      {
+         _addEvent.AddedObject.ShouldBeEqualTo(_refreshedFormula);
+      }
+
+
       [Observation]
       public void the_formula_that_was_replaced_is_removed_from_cache()
       {
          _initialConditionsBuildingBlock.FormulaCache.Contains(_replacedFormula).ShouldBeFalse();
       }
    }
 
+   public class When_updating_a_molecule_start_value_from_original_building_block_and_all_formulas_will_be_replaced_and_AddEvent_Only_executed_once : concern_for_initialConditions_With_formulas_for_Renaming
```

Comment:
This name probably should indicate that `added_event_should_not_be_called` instead.